Tests/SymbolKitTests/SymbolGraph/SemanticVersion/CodingTests.swift

```diff
@@ -0,0 +1,94 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import XCTest
+@testable import SymbolKit
+
+final class CodingTests:XCTestCase {
+    func testEncodingToJSON() throws {
+        func assertEncoding(_ version: SymbolGraph.SemanticVersion, to expectedJSONObject: String) throws {
+            let jsonEncoder = JSONEncoder()
+            jsonEncoder.outputFormatting = [.withoutEscapingSlashes, .sortedKeys]
+            let encodedVersion = try jsonEncoder.encode(version)
+            XCTAssertEqual(
+                String(data: encodedVersion, encoding: .utf8),
+                // Because Semantic Versioning 2.0.0 does not allow whitespace in identifiers, we can remove whitespace from the expected JSON object string.
```

Comment:
I think this is necessary in order to keep the tests not too complicated. Since whitespace is non-semantic in JSON (outside of strings), I think the comparison is as-is in a sense. Perhaps I can make this behaviour a bit more clear in the function naming. I'll think a bit more on it.